subsys/bluetooth/host/classic/conn_br.c

```diff
@@ -170,3 +170,107 @@ void bt_br_acl_recv(struct bt_conn *conn, struct net_buf *buf, bool complete)
 
 	net_buf_unref(buf);
 }
+
+int bt_conn_br_switch_role(const struct bt_conn *conn, uint8_t role)
+{
+	struct net_buf *buf = NULL;
+	struct bt_hci_cp_switch_role *cp;
+
+	CHECKIF(conn == NULL) {
+		LOG_DBG("conn is NULL");
+		return -EINVAL;
+	}
+
+	if (!bt_conn_is_type(conn, BT_CONN_TYPE_BR)) {
+		LOG_DBG("Invalid connection type: %u for %p", conn->type, conn);
+		return -EINVAL;
+	}
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_SWITCH_ROLE, sizeof(*cp));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	bt_addr_copy(&cp->bdaddr, &conn->br.dst);
+	cp->role = role;
+
+	return bt_hci_cmd_send_sync(BT_HCI_OP_SWITCH_ROLE, buf, NULL);
+}
+
+static int bt_conn_br_read_link_policy_settings(const struct bt_conn *conn,
+						uint16_t *link_policy_settings)
+{
+	int err;
+	struct net_buf *buf = NULL;
+	struct bt_hci_cp_read_link_policy_settings *cp;
+	struct bt_hci_rp_read_link_policy_settings *rp;
+	struct net_buf *rsp;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_READ_LINK_POLICY_SETTINGS, sizeof(*cp));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = conn->handle;
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_READ_LINK_POLICY_SETTINGS, buf, &rsp);
+	if (!err) {
+		rp = (void *)rsp->data;
+		*link_policy_settings = rp->link_policy_settings;
+	}
+
+	return err;
+}
+
+static int bt_conn_br_write_link_policy_settings(const struct bt_conn *conn,
+						 uint16_t link_policy_settings)
+{
+	struct net_buf *buf = NULL;
+	struct bt_hci_cp_write_link_policy_settings *cp;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_WRITE_LINK_POLICY_SETTINGS, sizeof(*cp));
+	if (!buf) {
+		return -ENOBUFS;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = conn->handle;
+	cp->link_policy_settings = link_policy_settings;
+
+	return bt_hci_cmd_send_sync(BT_HCI_OP_WRITE_LINK_POLICY_SETTINGS, buf, NULL);
+}
+
+int bt_conn_br_set_role_switchable(const struct bt_conn *conn, bool enable)
+{
+	int err;
+	uint16_t link_policy_settings;
+	bool is_enabled;
+
+	CHECKIF(conn == NULL) {
+		LOG_DBG("conn is NULL");
+		return -EINVAL;
+	}
+
+	if (!bt_conn_is_type(conn, BT_CONN_TYPE_BR)) {
+		LOG_DBG("Invalid connection type: %u for %p", conn->type, conn);
+		return -EINVAL;
+	}
+
+	err = bt_conn_br_read_link_policy_settings(conn, &link_policy_settings);
+	if (err) {
+		return err;
+	}
+
+	is_enabled = link_policy_settings & BT_HCI_LINK_POLICY_SETTINGS_ENABLE_ROLE_SWITCH;
+
+	if (enable == is_enabled) {
```

Comment:
To make it clear that you mean `&` and not `&&` I think it's better to put the operation inside parenthesis. Also, no empty line between the assignment and the test, please (this is also inconsistent with yourself, since you don't do it e.g. on line 261/262 either).